src/Microsoft.ML.Core/Properties/PublicKey.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !PRIVATE_CONTRACTS
+// This namespace contains the PublicKey generally used in ML.NET project settings
+namespace Microsoft.ML
+#else
+// CpuMath module has its own PublicKey for isolating itself from Microsoft.ML.Core
+// CpuMath has its own BestFriend defined in Microsoft.ML.Hidden.
+using Microsoft.ML.Hidden;
+namespace Microsoft.ML.Runtime.Internal.CpuMath
```

Comment:
Why not just make this `namespace Microsoft.ML.Hidden` like the other types have? Then you don't need the `using`, and it is all consistent.

It feels odd for this one place to use a `CpuMath` namespace, but `Contracts` and `BestFriend` to use `Hidden`.

Reply:
Things particularly for CpuMath are moved to Microsoft.ML.Runtime.Internal.Cpu now.

---
In reply to: [236737148](https://github.com/dotnet/machinelearning/pull/1724#discussion_r236737148) [](ancestors = 236737148)